templates/common/_base/files/configure-ovs-network.yaml

```diff
@@ -0,0 +1,134 @@
+    iface_ip=""
+    if ip -4  addr show ${iface} |grep -Eo "([0-9]{1,3}[\.]){3}[0-9]{1,3}"; then
+      iface_ip=$(ip -4  addr show ${iface} | grep -Eo "([0-9]{1,3}[\.]){3}[0-9]{1,3}" | head -n 1)
+    fi
+
+    # create bridge
+    if ! nmcli connection show br-ex &> /dev/null; then
+      nmcli c add type ovs-bridge conn.interface br-ex con-name br-ex 802-3-ethernet.mtu ${iface_mtu} 802-3-ethernet.cloned-mac-address ${iface_mac}
+    fi
+
+    # find default port to add to bridge
+    if ! nmcli connection show ovs-port-phys0 &> /dev/null; then
+      nmcli c add type ovs-port conn.interface ${iface} master br-ex con-name ovs-port-phys0
+    fi
+
+    if ! nmcli connection show ovs-port-br-ex &> /dev/null; then
+      nmcli c add type ovs-port conn.interface br-ex master br-ex con-name ovs-port-br-ex
+    fi
+
+    if nmcli conn show --active | grep ${iface} | grep "Wired" ; then
+      old_conn=$(nmcli conn show --active | grep ${iface} | egrep '[0-9a-f]{8}-([0-9a-f]{4}-){3}[0-9a-f]{12}' -o)
+      nmcli conn down $old_conn
+    fi
+
+    if ! nmcli connection show ovs-if-phys0 &> /dev/null; then
+      nmcli c add type 802-3-ethernet conn.interface ${iface} master ovs-port-phys0 con-name ovs-if-phys0 \
+        connection.autoconnect-priority 100 802-3-ethernet.mtu ${iface_mtu}
+    fi
+
+    if ! nmcli connection show ovs-if-br-ex &> /dev/null; then
+      nmcli c add type ovs-interface slave-type ovs-port conn.interface br-ex master ovs-port-br-ex con-name \
+        ovs-if-br-ex 802-3-ethernet.mtu ${iface_mtu}
+    fi
+
+    # setting MTU on ovs iface does not seem to propagate to OVS
+    # https://gitlab.freedesktop.org/NetworkManager/NetworkManager/-/issues/206
+    counter=0
+    until ovs-vsctl set Interface br-ex mtu_request=${iface_mtu}; do
+      if [ $counter -gt 20 ]; then
+        echo "WARN: failed to set MTU in OVS"
+        break
+      fi
+      echo "br-ex interface not yet present in OVS"
+      counter=$((counter+1))
+      sleep 1
+    done
+
+    # bounce connection to make sure interface gets right MTU
+    # nmcli conn up cmd may return error if dhcp fails (see bug 483 below)
+    set +e
+    nmcli conn down ovs-if-br-ex
+    sleep 5
+    nmcli conn up ovs-if-br-ex
+
+    # need to ensure we got the right DHCP IP
+    # there is a bug in NM where DHCP request may be sent with wrong MAC for OVS iface
+    # https://gitlab.freedesktop.org/NetworkManager/NetworkManager/-/issues/483
+    # workaround by bouncing networking until it picks the right iface if the original iface was ipv4
+    if [ ! -z "$iface_ip" ]; then
+      echo "checking iface IP matches original DHCP leased IP"
+      # sleep for a few seconds to allow DHCP to finish
+      sleep 10
+      counter=0
+      while [ $counter -lt 30 ]; do
+        new_iface_ip=$(ip -4  addr show br-ex |grep -Eo "([0-9]{1,3}[\.]){3}[0-9]{1,3}" | head -n 1)
+        if [ "$new_iface_ip" = "$iface_ip" ]; then
+          echo "DHCP successful on OVS iface"
+          exit 0
```

Comment:
Maybe we could check connectivity to the gateway for extra points and rollback if we cant.

Reply:
this whole thing should go away when the bug is fixed, which is now verified:
https://bugzilla.redhat.com/show_bug.cgi?id=1852106